include/aoscene.h

```diff
@@ -23,11 +54,37 @@ class AOScene : public QLabel {
   // no "combo" to speak of
   void combo_resize(int w, int h);
 
+  // Return the frame delay adjusted for speed
+  int get_frame_delay(int delay);
+
+  QStringList network_strings;
+
+  QString m_char;
+  QString m_emote;
+
 private:
-  QWidget *m_parent;
-  QMovie *m_movie;
   AOApplication *ao_app;
-  QString last_image;
+
+  QVector<QPixmap> movie_frames;
+  QVector<int> movie_delays;
+
+  // Effects such as sfx, screenshakes and realization flashes are stored in
+  // here. QString entry format: "sfx^[sfx_name]", "shake", "flash". The program
+  // uses the QVector index as reference.
+  QVector<QVector<QString>> movie_effects;
+
+  QTimer *preanim_timer;
+  QTimer *shout_timer;
+  QTimer *effect_timer;
+  QTimer *ticker;
+  QString last_path;
+  QImageReader *m_reader = new QImageReader();
```

Comment:
not Java - use static allocation, or allocate in the constructor with `this` as the QObject's parent

Reply:
Not quite sure how to do something like that - this particular snippet is pretty much verbatim from AOCharMovie so I didn't write it

Reply:
Worked out over Discord